lib/Doctrine/DBAL/Types/DecimalType.php

```diff
@@ -30,6 +34,10 @@ public function getSQLDeclaration(array $column, AbstractPlatform $platform)
      */
     public function convertToPHPValue($value, AbstractPlatform $platform)
     {
+        if (PHP_VERSION_ID >= 80100 && is_float($value)) {
```

Comment:
Should it be commented with description - why is this `if` (with PHP version check) and casting added?

Reply:
Sure why not. Do you want to open a PR with a good comment? Maybe link to this PR.

Reply:
Yes, will do 
Added to today's schedule